src/main.rs

```diff
@@ -39,23 +41,49 @@ struct Args {
     #[arg(long)]
     timeout_sec: u64,
 
+    // How long to cool off before runs
+    #[arg(long, default_value_t = 0)]
+    cooloff_sec: u64,
+
     #[arg(long, default_value_t = 5)]
     times_to_run: u32,
 
-    #[arg(long)]
+    #[arg(long, default_value_t = String::new())]
     cgroup_path: String,
 }
 
 // Move the pid to the cgroup
-fn move_to_cgroup(pid: u32, context: &RunContext) -> Result<()> {
-    let procs_file_path = format!("{}/cgroup.procs", context.cgroup_path);
-    let pid_string = pid.to_string();
-
+fn drop_caches() -> Result<()> {
     std::fs::OpenOptions::new()
         .write(true)
         .append(true)
-        .open(procs_file_path)?
-        .write_all(pid_string.as_bytes())?;
+        .open("/proc/sys/vm/drop_caches")?
+        .write_all("3".as_bytes())?;
+
+    Ok(())
+}
+
+fn cooloff(context: &RunContext) -> Result<()> {
+    if context.cooloff>0 {
+        debug!("Dropping caches and cool-off delay {0}s", context.cooloff);
+        drop_caches()?;
+        thread::sleep(time::Duration::from_secs(context.cooloff));
+    }
+    Ok(())
+}
```

Comment:
How much improvement is this really bringing on a shared VM though?

Reply:
How can anyone know. But even if it weren't a shared host, some solutions will be IO intensive (pegging all cores at 100%) which may lead to throttling. So, that's why I add it on my runs (cooloff = 15s)

I already said that running this on dedicated hardware under your control is required:

> Of course the only *real* way is to run this on dedicated bare metal and
disable CPU scaling altogether